connectors/byoc.py

```diff
@@ -398,7 +400,7 @@ def next_sync(self):
         return next_run(self.scheduling["interval"])
 
     async def _sync_starts(self):
-        job = SyncJob(self.id, self.client)
+        job = SyncJob(connector_id=self.id, elastic_index=self.index)
```

Comment:
Should pass a `SyncJobIndex` instance here?

Reply:
It doesn't make sense to do it now because `_sync_starts` will be moved out from `Connector` as we are decoupling scheduling and consuming workloads (I guess you are working on it). 

I'm trying to move in small steps so this is a kind of mock.